The patch fixes build failure of form:
     ../src/alpha/osf.S:298:2: error: #error "osf.S out of sync with ffi.h"
Upstream fixed the bug in a more invasive way
but didn't have releases since 3.2.1.

The patch is taken from Gentoo:
https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-libs/libffi/files/libffi-3.2.1-complex_alpha.patch

--- libffi-3.2.1/src/alpha/osf.S	2015-01-16 10:46:15.000000000 +0100
+++ libffi-3.2.1/src/alpha/osf.S	2015-01-16 10:46:24.000000000 +0100
@@ -279,6 +279,7 @@
 	.gprel32 $load_64	# FFI_TYPE_SINT64
 	.gprel32 $load_none	# FFI_TYPE_STRUCT
 	.gprel32 $load_64	# FFI_TYPE_POINTER
+	.gprel32 $load_none	# FFI_TYPE_COMPLEX
 
 /* Assert that the table above is in sync with ffi.h.  */
 
@@ -294,7 +295,8 @@
 	|| FFI_TYPE_SINT64 != 12	\
 	|| FFI_TYPE_STRUCT != 13	\
 	|| FFI_TYPE_POINTER != 14	\
-	|| FFI_TYPE_LAST != 14
+	|| FFI_TYPE_COMPLEX != 15       \
+	|| FFI_TYPE_LAST != 15
 #error "osf.S out of sync with ffi.h"
 #endif
 
